src/buffer.c

```diff
@@ -947,6 +945,9 @@ compact_buffer (struct buffer *buffer)
     }
 }
 
+extern Lisp_Object Fkill_buffer(Lisp_Object buffer_or_name);
```

Comment:
This should not be needed. The compilation process is supposed to generate this.